### ISSUE TITLE
[DataGrid] Unable to override default props/styling for GridActionsCellItem

### ISSUE THREAD

**Issue by Nitzperetz** [mui/mui-x#7619]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Steps to reproduce 🕹

Overriding theme for MuiDataGrid was added in v5.0.0, however not all DataGrid Types were added like - GridActionsCellItem

```
{

   overrides: {
     // MuiDataGrid override works:
     MuiDataGrid: {
         root: {
           ...
         }
      },   
     // GridActionsCellItem override doesn't work:
     GridActionsCellItem: {
         root: {
           ...
         }
      },        
   },
}

### Current behavior 😯

Unable to override theme for GridActionsCellItem.

### Expected behavior 🤔

Override GridActionsCellItem defaultProps and styling.

### Context 🔦

Overriding the theme in MUI is a feature I use in almost every component in order to align UX/UI. The only component I wasn't able to customize is the cell item for now.

Openning this issue as requested [here](https://github.com/mui/mui-x/issues/2450#issuecomment-1387345876).

### Your environment 🌎

Browser is not relevant.
Having issues with running the command. Will update the issue once it's fixed.

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>


### Order ID 💳 (optional)

_No response_

**Comment by cherniavskii** [comment]
Thanks for opening the issue!

For `styleOverrides`, I believe it should look like this:
```tsx
const theme = createTheme({
  components: {
    MuiDataGrid: {
      styleOverrides: {
        actionsCellItem: {
          // ...
        }
      }
    }
  }
});
```
It doesn't work though, it's a bug.

For the `defaultProps`, from what I can tell you can only pass the props to the root component (`DataGrid`):
```tsx
const theme = createTheme({
  components: {
    MuiDataGrid: {
      defaultProps: {
        // ...
      }
    }
  }
});
```

**Comment by LukasTy** [comment]
Hello @Nitzperetz 
Data grid package has a somewhat different theme overriding approach to date-pickers if you are comparing the two by any chance.
As @cherniavskii mentioned, [DataGrid only allows](https://github.com/mui/mui-x/blob/master/packages/grid/x-data-grid/src/themeAugmentation/props.ts) overriding both props and styles on the root element itself.
Do you have any particular use case where you would want certain `defaultProps` on `GridActionsCellItem`?

As for the style overrides, you can find the possible classes you can target [here](https://github.com/mui/mui-x/blob/master/packages/grid/x-data-grid/src/constants/gridClasses.ts).
[Here is an example](https://codesandbox.io/s/nice-chaum-sp69jm?file=/demo.tsx) of how it's possible to override `actionsCell` styles. On this you are correct. There is no `actionsCellItem` option, but as you can see—there is already a PR created by @cherniavskii to add it in.

You might also expect that applying the following change should produce the same result.
```diff
 const theme = createTheme({
   components: {
     MuiDataGrid: {
       styleOverrides: {
-        root: {
+        actionsCell: {
-          [`.${gridClasses.actionsCell}`]: {
             backgroundColor: "red"
-           }
         }
       }
     }
   }
 });
```
But that is not the case, because the given component isn't a `styled` one and does not support direct overrides.
Which is probably an issue on our side (@mui/xgrid), because based on typing support—you could expect that to work as well.

As with anything, it's best to explain your case with as clear examples as possible. It's always perfect if you provide a codesandbox with what is the end goal you are trying to achieve.

**Comment by cherniavskii** [comment]
@LukasTy I think this should work:
```tsx
styleOverrides: {
  actionsCell: {
    backgroundColor: "red"
  }
}
```

The same way as this works: https://codesandbox.io/s/tender-bird-vikzq5?file=/demo.tsx
```tsx
styleOverrides: {
  row: {
    backgroundColor: "red"
  }
}
```

> But that is not the case, because the given component isn't a `styled` one and does not support direct overrides

I don't think it's related to `styled`.
Those overrides only work if style overrides are handled in GridRootStyles - see https://github.com/mui/mui-x/pull/7635/files#r1084306896

**Comment by LukasTy** [comment]
> > But that is not the case, because the given component isn't a `styled` one and does not support direct overrides
> 
> I don't think it's related to `styled`.

Yes, sorry for not expressing myself correctly. For named style overrides to work they need to be consumed by any `styled` components' `overridesResolver`.
[Here is a working example](https://codesandbox.io/s/busy-tharp-0xrly1?file=/demo.tsx) with your mentioned way of overriding it on your PR.

**Comment by cherniavskii** [comment]
> For named style overrides to work they need to be consumed by any styled components' overridesResolver.

Makes sense 👍
In this case, the `GridRootStyles`'s overridesResolver consumes those overrides

**Comment by Nitzperetz** [comment]
Thank you all for replying!

> As @cherniavskii mentioned, [DataGrid only allows](https://github.com/mui/mui-x/blob/master/packages/grid/x-data-grid/src/themeAugmentation/props.ts) overriding both props and styles on the root element itself.
> Do you have any particular use case where you would want certain `defaultProps` on `GridActionsCellItem`?

I am in need to align GridActionsCellItem component across the app. 
For styling I just wanted to change fontSize.
For PropsOverride the need is to set `showInMenu` to be true by default.

**Comment by dBianchii** [comment]
Also was not able to do it. I nee dto override the default props of GridActionsCellItem, not the styles
